keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/DoubleVertex.java

```diff
@@ -185,11 +181,11 @@ public final DualNumber getDualNumber() {
         return dualNumbers.get(this);
     }
 
-    private Set<Vertex> parentsThatDualNumberIsNotCalculated(Map<Vertex, DualNumber> dualNumbers, Set<Vertex> parents) {
-        Set<Vertex> notCalculatedParents = new HashSet<>();
+    private Set<DoubleVertex> parentsThatDualNumberIsNotCalculated(Map<Vertex, DualNumber> dualNumbers, Set<Vertex> parents) {
+        Set<DoubleVertex> notCalculatedParents = new HashSet<>();
         for (Vertex<?> next : parents) {
-            if (!dualNumbers.containsKey(next)) {
-                notCalculatedParents.add(next);
+            if (!dualNumbers.containsKey(next) && next instanceof DoubleVertex) {
+                notCalculatedParents.add((DoubleVertex) next);
```

Comment:
What do you guys think about logging a warning here saying that we have skipped the dual number due to the parent not being a double?

Reply:
why don't you want to throw an exception? i.e. what makes this "expected" behaviour?

Reply:
I think this bit of code should differentiate what it can and some higher level code (something that can better respond to there being non-differentiable network paths) should worry about if the bit that is differentiated is sufficient for it's needs.